matijapretnar/eff
# Dune tests

Ported old regression tests to dune

Each file to be tested comes with two dune rules, one for invocation and one that aliases it to `dune runtest`. Rules are autogenerated by `generate_dune_rules.ml` and saved in file `dune.inc`.

Run tests with:
`make test` or `dune runtest`

To add new test case to existing test folders:
- Create test file `text_xyz.eff` and reference file `text_xyz.eff.ref`
- Run `dune runtest` to generate updated `dune.inc` file and `dune promote` to update it


To create a new folder with tests
- Create new folder somewhere under `tests`
- Copy `dune` file from existing test folders and add empty `dune.inc` file. Fix path to `generate_dune_rules` and adjust configurations for new test generation. You can only configure expected exit code for now.
- Add test cases
